generate.py

```diff
@@ -49,11 +50,20 @@ def getSources(root="./src"):
     return sources
 
 
-def generate(src="src", target="output", format="png", style="modern-blue"):
+def generate(src="src", target="output", format="png", style="modern-blue",
+             rebuild=False):
     for f in getSources(src):
         src_path = os.path.dirname(f)
         src_file = os.path.basename(f)
         target_path = target + src_path[src_path.index('/'):]
+
+        work_path = os.getcwd() + "/"
+        output_file_name = src_file[:src_file.rindex('.') + 1] + format
+        target_full_path = work_path + target_path + "/" + output_file_name
```

Comment:
You don't need get absolute path again, you can use `os.path.abspath` or `os.path.realpath` to convert to absolute path.

```python
abspath = os.path.abspath(target_path + output_file)
```